### ISSUE TITLE
Improvements for ion-back-button (disabled and text only for ios)

### ISSUE THREAD

**Issue by rafalschmidt97** [ionic-team/ionic-framework#17279]
# Feature Request

**Ionic version:**

[x] **4.0.0**

**Describe the Feature Request**

1. I would love to see **disabled** attribute for back button (sometimes we need to block actions for example when we wait for response - would be cool to block it). For now i just use **hidden** attribute but it looks weird when it disappears unexpectedly.

2. I need to translate "Back" text for ios. On android defaultly there is just an arrow (without text). But when i use attribute **text** it shows also text on android which I don't need.

![screenshot 2019-01-26 at 19 48 27](https://user-images.githubusercontent.com/14861637/51792191-43617f80-21ae-11e9-8f4f-1008c430ec8a.png)

**Describe Preferred Solution**

1. Change colour of icon and text and disable action.
2. Maybe a new attribute like **iosText** to change "Back" for ios. I don't need it on android.

`[iosText]="'Back' | translate"`

Have a nice day 😀

**Comment by arnold-kenaci** [comment]
You can change the text displayed for back button at the declaration of your root Ionic Module by passing the backButtonText attribute. For example: 
```@NgModule({   
 declarations: [
    MyApp,
    Home   
 ],   
 imports: [
    IonicModule.forRoot(MyApp, {
      tabsPlacement: 'top',
      backButtonText: 'Back'
     })],
 bootstrap: [IonicApp],
 entryComponents: [
    MyApp,
    Home   ],
 providers: [MyService]
})

**Comment by christianguevara** [comment]
Meanwhile to hide the text in android you can use in variables.scss

```
  .button-text.sc-ion-back-button-md {
    display: none;
  }
```

**Comment by rafalschmidt97** [comment]
@christianguevara that's a clever workaround 😂

**Comment by biesbjerg** [comment]
1) [A `disabled` attribute actually does already exist](https://github.com/ionic-team/ionic/blob/master/core/src/components/back-button/back-button.tsx#L40) for the back button component, but it doesn't look like it's exposed to Angular.

2) Regarding the back button text, you could do this in `AppComponent`'s ' `ngOnInit` (which is also more DRY than defining the button text everywhere) :

```ts
this.translateService.onLangChange.subscribe((event: LangChangeEvent) => {
	this.config.set('backButtonText', this.platform.is('ios') ? this.translateService.instant('Back') : '');
});
```

**Comment by rafalschmidt97** [comment]
@biesbjerg - thanks a lot! 🚀

**Comment by NiklasMerz** [comment]
`this.config.set` is deprecated and that's why there should still be an option to translate the back button text in a simple way.

**Comment by carstenbaumhoegger** [comment]
@christianguevara thank's for your solution!
I'm using Ionic 5 and it doesn't work anymore because they moved more components over to shadow dom. Does anyone have an updated solution for the problem that the text should only be shown when running on iOS?

**Comment by rafalschmidt97** [comment]
> @christianguevara thank's for your solution!
> I'm using Ionic 5 and it doesn't work anymore because they moved more components over to shadow dom. Does anyone have an updated solution for the problem that the text should only be shown when running on iOS?

In my case the problem doesn't exist as long as I'm using .config (provided by @biesbjerg) to set up the text instead of text="xyz". @NiklasMerz said it's deprecated but it works.

**Comment by liamdebeasi** [comment]
Hi everyone,

There are multiple requests here, so I will provide an update for each one.

1. **There should be a way to disable the `ion-back-button` and have it be visually disabled.**
After discussing this with the team, this is not a pattern we are looking to encourage. Disabled back buttons are typically not done in native mobile apps. Additionally, disabling the back button has implications for the swipe to go back gesture as well as the browser back button which we think will lead to a confusing user experience. If you wish to prevent a user from leaving a page, we recommend using routing guards in your JS Framework's router integration instead.


2. **There should be a way to localize the `ion-back-button` text on iOS without showing text on Android.**
This is something that is already possible with the current Ionic API. Developers can use the `isPlatform` function to only apply the localized text when running on an iOS device.

Here is an example with Angular:

```typescript
import { isPlatform, IonicModule } from '@ionic/angular';

@NgModule({
  ...
  imports: [
    BrowserModule,
    IonicModule.forRoot({
      backButtonText: isPlatform('ios') ? 'My Localized Text Here' : undefined
    }),
    AppRoutingModule
  ],
  ...
})
```

This pattern also aligns with DRY principles and should address the concern noted in https://github.com/ionic-team/ionic-framework/issues/17279#issuecomment-937838097. Developers can also do this on a per-instance basis by using the same `isPlatform` function with the `text` property on `ion-back-button`.

`isPlatform` docs for Angular: https://ionicframework.com/docs/angular/config#per-platform-config
`isPlatform` docs for React: https://ionicframework.com/docs/react/config#per-platform-config
`isPlatform` docs for Vue: https://ionicframework.com/docs/vue/config#per-platform-config

--------

To summarize, the disabled back button functionality is not something we are looking to implement, and the per-platform back button text functionality is already possible in Ionic. As a result, I am going to close this issue. Thank you!
